app/build.gradle

```diff
@@ -97,29 +97,35 @@ android {
 }
 
 dependencies {
+    def composeBom = platform(libs.compose.bom)
     implementation fileTree(include: ['*.jar'], dir: 'libs')
-    implementation rootProject.ext.picasso
-    implementation rootProject.ext.accompanistFlowLayout
-    implementation rootProject.ext.androidXLifecycleKtx
-    implementation rootProject.ext.composeActivity
-    implementation rootProject.ext.kotlinReflection
-    implementation rootProject.ext.stetho
-    implementation rootProject.ext.destinationsCore
-    implementation rootProject.ext.accompanistPager
-    testImplementation rootProject.ext.junit
-    debugImplementation rootProject.ext.composeTestManifest
-    androidTestImplementation rootProject.ext.composeFoundation
-    androidTestImplementation rootProject.ext.testCoreKtx
-    androidTestImplementation rootProject.ext.junitKtx
-    androidTestImplementation rootProject.ext.junitAndroid
-    androidTestImplementation rootProject.ext.espressoCore
-    androidTestImplementation rootProject.ext.espressoContrib
-    androidTestImplementation rootProject.ext.testRules
-    androidTestImplementation rootProject.ext.coroutinesTest
-    androidTestImplementation rootProject.ext.composeTestJUnit
-    androidTestImplementation rootProject.ext.facebookSnapshotTestingCore
+
+    implementation composeBom
+    implementation libs.square.picasso
+    implementation libs.accompanist.flowLayout
+    implementation libs.androidx.lifecycleKtx
+    implementation libs.compose.activity
+    implementation libs.kotlin.reflection
+    implementation libs.destinations.core
+    implementation libs.accompanist.pager
+    testImplementation libs.test.junit
+
+    debugImplementation composeBom
```

Comment:
do you actually have to re-declare it for everything? in the app we actually add the bom to all dependencies and declare it as an array instead, so you don't have to manually add it

Reply:
It seems so:
https://discuss.gradle.org/t/applying-a-bom-platform-to-all-configurations/30621

You were able to do this because it was from the code. Toml just tells Gradle how to generate dependency providers and it cannot do this. Here's the example of the same from Compose website:
https://developer.android.com/jetpack/compose/bom/bom#does_the_bom_work_with_version_catalogs